backend/devices/bitbox02/keystore_simulator_test.go

```diff
@@ -412,12 +419,48 @@ func makeTx(t *testing.T, device *Device, recipient *maketx.OutputInfo) *btc.Pro
 		GetPrevTx: func(chainhash.Hash) (*wire.MsgTx, error) {
 			return prevTx, nil
 		},
+		GetAddress: func(scriptHashHex blockchain.ScriptHashHex) *addresses.AccountAddress {
+			for _, address := range addrsInDifferentAccount {
+				if address.PubkeyScriptHashHex() == scriptHashHex {
+					return address
+				}
+			}
+			return nil
+		},
 		Signatures: make([]*types.Signature, len(txProposal.Transaction.TxIn)),
 		FormatUnit: coinpkg.BtcUnitDefault,
 	}
 
 }
 
+func TestSimulatorSignBTCTransactionSendSelfDifferentAccount(t *testing.T) {
```

Comment:
Maybe also add a unit test that covers sending to an address of a different keystore (which I suspect does not currently work as expected).

Reply:
@benma I'm not sure that would give us any meaningful coverage; the logic to exclude different keystores is in the `GetAddress` function, which we mock here, so we would basically only change the mock function.

Another idea was to extract the `getAddressCallback` function into a separate fn and test it. However, the reason this also not works is that the functions needs to perform a type cast to `btc.Account` (as it needs to call `GetAddress`, which is not parts of the `account` interface) so this won't work with the mock type generated by `moq`. What do you think? I can't see a valid way to add a unit test for this but open to suggestions!